compiler/rustc_builtin_macros/src/format.rs

```diff
@@ -1270,6 +1275,24 @@ pub fn expand_preparsed_format_args(
                 e.span_label(fmt_span.from_inner(InnerSpan::new(span.start, span.end)), label);
             }
         }
+        if err.should_be_replaced_with_positional_argument {
+            let captured_arg_span =
+                fmt_span.from_inner(InnerSpan::new(err.span.start, err.span.end));
+            let positional_args = args.iter().filter(|arg| !arg.named).collect::<Vec<_>>();
+            let mut suggestions = vec![(captured_arg_span, positional_args.len().to_string())];
+            if let Ok(arg) = ecx.source_map().span_to_snippet(captured_arg_span) {
```

Comment:
Let's maybe keep this check enclose the whole suggestion? So if we fail to obtain the `captured_arg_span` snippet, we don't display the suggestion at all (instead of a partial suggestion).